src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -90,7 +90,7 @@ protected override void CheckBase(DiagnosticBag diagnostics)
             }
 
             // you need to know all bases before you can ask this question... (asking this causes a cycle)
-            if (this.IsGenericType && !localBase.IsErrorType() && this.DeclaringCompilation.IsAttributeType(localBase))
+            if (this.DeclaringCompilation.LanguageVersion < LanguageVersion.CSharp8 && this.IsGenericType && !localBase.IsErrorType() && this.DeclaringCompilation.IsAttributeType(localBase))
```

Comment:
We'll want to test a C# 7 compiler trying to consume a generic attribute. What happens in such a case?

Reply:
Consuming generic attributes not belong to the compiler it's depend on the runtime

Reply:
It would be good to know if something bad happens (such as a compiler crash) in this scenario (to test manually):
1. A C# 8 compiler produces a library with `MyAttribute<T>`.
2. A C# 7 compiler tries to compile code with `[MyAttribute]` and referencing the above library.

Reply:
I tested it manually the compiler not crash it returns the correct error
also there is a test for that scenario look at GenericAttributeTypeFromILSource